Tests/SonarQube.TeamBuild.PreProcessor.Tests/DataModel/ServerDataModel.cs

```diff
@@ -34,14 +35,20 @@ public ServerDataModel()
 
         public Repository AddRepository(string repositoryKey, string language)
         {
-            Repository repo = new Repository(repositoryKey, language);
+            Repository repo = this.FindRepository(repositoryKey, language);
+            Assert.IsNull(repo, "A repository already exists. Key: {0}, language: {1}", repositoryKey, language);
+
+            repo = new Repository(repositoryKey, language);
```

Comment:
Minor refactoring to use the new "FindXXX" methods, and to fail with an obvious message if an attempt is made to add duplicates (attempting to add a duplicate would have failed before, but not in such an obvious way).